app/controllers/HomeController.scala

```diff
@@ -101,7 +85,7 @@ class HomeController @Inject() (trafficLightService: TrafficLightService,
             test
           }
           case None => {
-            save(Map(newTrafficLight.id -> newTrafficLight)) // create a new light
+            trafficLightService.save(newTrafficLight) // create a new light
```

Comment:
Saving a `Map` is not necessary (and may be hard to implement if you are using a database). You can just save the `TrafficLight` itself.

Reply:
Ah yeah true...